### PR TITLE
Added constructors.md

### LINKED ISSUE #4247: [Entry] Constructors: Create new concept entry `constructors.md`
### Reason/inspiration (optional)

We would like a new term entry on the `Constructors` term for [Dart](https://www.codecademy.com/resources/docs/dart). The entry should go under the following path `docs/content/dart/concepts/constructors/contructors.md`.

The entry should describe the term and how it is used within Dart. Also include: 
- A section that explains constructors in Dart.

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by vjurcutiu**
### Description

Added the constructors.md file, which covers the concept of constructors in Dart.

### Issue Solved

Closes #4247

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
→

### CODE REVIEW

File: dart/concepts/constructors/constructors.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+A **constructor** is a special method that initializes an instance of a class. This method is called automatically whenever an object is created. All objects have a default constructor (whether declared or not), but certain languages allow for the creation of custom constructor logic that will determine the object behavior on instantiation.
+
+In Dart, there are six constructor types:
+
+- Default constructors
+- Generative constructors
+- Named constructors
+- Redirecting constructors
+- Constant constructors
+- Factory constructors
```
Comment (dakshdeepHERE) [inline dart/concepts/constructors/constructors.md:25]:
```suggestion
In Dart, there are six constructor types:

- `Default constructors`
- `Generative constructors`
- `Named constructors`
- `Redirecting constructors`
- `Constant constructors`
- `Factory constructors`
```

File: dart/concepts/constructors/constructors.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+void main() {
+    var testPixel1 =Pixel.generate('white');
+    var testPixel2 = GreenPixel.generate('green');
+  }
+```
+
+## Super parameters and the initializer list
+
+In order to better deal with constructor parameters and superclasses, Dart offers two main tools: super parameters and the initializer list.
+Super parameters will forward parameters to the declared superclass constructor. They help avoid having to manually pass each parameter in the super invocation of a constructor.
```
Comment (dakshdeepHERE) [inline dart/concepts/constructors/constructors.md:139]:
```suggestion
To better deal with constructor parameters and superclasses, Dart offers two main tools: superparameters and the initializer list. Superparameters will forward parameters to the declared superclass constructor. They help avoid having to manually pass each parameter in the super invocation of a constructor:
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Nice entry, for now I have few comments for you to work on @vjurcutiu

**Review by Helfese — APPROVED** [review]

**Review by ishg-153 — APPROVED** [review]
@vjurcutiu Great Entry! Looks ready to merge.

**Pull request merged**